### ISSUE TITLE
Adding support for lens type classes

### ISSUE THREAD

**Issue by tmcgilchrist** [hedgehogqa/haskell-hedgehog-classes#1]
Would you consider adding or accepting a PR to add lens type class support?
eg Prism, Iso

**Comment by chessai** [comment]
Yes, I would.

**Comment by andrewthad** [comment]
Laws for all lawful typeclasses are welcome so long as they come with a manual flag that can be used to disable their them (the flags are needed because `primitive`'s test suite uses `quickcheck-classes`, and the extra deps would make travis take a long time).

**Comment by chessai** [comment]
Yes, thanks for mentioning that @andrewthad. However I think you thought that this was quickcheck-classes when you made that comment.

The same does apply here though: All laws from typeclasses external to base need: a flag that toggled their status as a build dependency, a cpp-option "HAVE_${PACKAGENAME}", and for the library/test suite only to build the Laws/tests for the associated typeclass (using the CPP).

**Comment by andrewthad** [comment]
Yep, my bad. I was thinking of the wrong repo.

**Comment by kozross** [comment]
It's be nice to have this for the `optics` versions too.

**Comment by jkachmar** [comment]
It looks like there's a project that implements some `hedeghog` checkers for the `optics` library: https://github.com/typeclasses/hedgehog-optics/blob/master/hedgehog-optics/src/Hedgehog/Optics.hs
